hack/extract.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang-collections/collections/stack"
+)
+
+var funcs map[string]struct{}
+var funcStack *stack.Stack
+var translations map[string]interface{}
+
+func main() {
+	translations = make(map[string]interface{})
+	paths := []string{"../cmd", "../pkg"}
+	//paths := []string{"../cmd/minikube/cmd/delete.go"}
+	//paths := []string{"../pkg/minikube/cluster/cluster.go"}
+	funcs = map[string]struct{}{"Translate": {}}
+	funcStack = stack.New()
+	for f := range funcs {
+		funcStack.Push(f)
+	}
+
+	fmt.Println("Compiling translation strings...")
+	for funcStack.Len() > 0 {
+		f := funcStack.Pop().(string)
+		//fmt.Printf("-----%s------\n", f)
+		for _, root := range paths {
+			err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+				if shouldCheckFile(path) {
+					return inspectFile(path, f)
+				}
+				return nil
+			})
+
+			if err != nil {
+				panic(err)
+			}
+		}
+	}
+
+	err := writeStringsToFiles()
+
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Done!")
+}
+
+func writeStringsToFiles() error {
+	translationsFiles := "../pkg/minikube/translate/translations"
+	e := filepath.Walk(translationsFiles, func(path string, info os.FileInfo, err error) error {
+		if info.Mode().IsDir() {
+			return nil
+		}
+		fmt.Printf("Writing to %s\n", filepath.Base(path))
+		var currentTranslations map[string]interface{}
+		f, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
```

Comment:
Since there are 5 possible errors returned by writeStringsToFiles, use errors.Wrap() to disambiguate them.

Here, and in any other function where there are multiple possible error scenarios.